lib/puppet/functions/extlib/ip_to_reverse.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'ipaddr'
+
+# @summary Returns the reverse of an IP address
+#
+# Not actual query is done to the DNS server, it only calculate the record
+# that could be added to a DNS zone file.
+Puppet::Functions.create_function(:'extlib::ip_to_reverse') do
+  # @param ip
+  #   IPv4 or IPv6 address
+  #
+  # @example Calling the function
+  #   extlib::ip_to_reverse('192.0.2.0')
+  dispatch :ip_to_reverse? do
```

Comment:
What's with the `?` at the end of the dispatch name?

Reply:
A bug in https://github.com/voxpupuli/puppet-extlib/blob/master/lib/puppet/functions/extlib/is_in_cidr.rb#L14 it seems :smiley:

Reply:
I guess it's only a _bug_ if the function name ends with a `?`.  In ruby itself, it's quite common for methods that return booleans to end with a `?`.  `ip_to_reverse` doesn't though, so the `?` is out of place here.